elastic/kibana
# [Alerting] remove ruleTaskTimeout and use config.execution.timeout instead

fixes: #124807

## Summary

This PR intends to remove `ruleTaskTimeout` config and use `config.execution.timeout` key instead of it.
The new key `config.execution.timeout` can be modified by the users globally or per rule type.

## To Verify
- Add a sleep function into the task runner code such as: `await new Promise(r => setTimeout(r, 60000));`
   That would delay your rule execution for a minute.
- Add the below config to your kibana.yml and run kibana
- Create two rules (e.g. Index Threshold and Elasticsearch Query ) and wait your rules to be executed.

 Index Threshold should timeout in 10s and the other one in 30s
 
```
xpack.alerting.rules.execution:
    timeout: 30s
    ruleTypeOverrides:
      - id: '.index-threshold'
         timeout: 10s
```


### Checklist

Delete any items that are not applicable to this PR.
